File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<Integer>();

        for (int i = 0; i < progresses.length; i++) {
            double days = (100 - progresses[i]) / (double) speeds[i];
            int daysUp = (int) Math.ceil(days);

            int j = i + 1;
            for (; j < progresses.length; j++)
                if (progresses[j] + daysUp * speeds[j] < 100)
                    break;

            answer.add(j - i);
            i = j - 1;
        }
        
        int[] answerArray = new int[answer.size()];
        
        for(int i = 0; i<answer.size(); i++) {
            answerArray[i] = answer.get(i);
        }
        
        return answerArray;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque
import math

def solution(progresses, speeds):    
    answer = []
    
    speedsQueue = deque(speeds)
    progressQueue = deque(progresses)
    days = 0
    
    while progressQueue:
        currentProgress = progressQueue[0] + speedsQueue[0] * days
        
        if currentProgress < 100:
            remainDays = 100 - currentProgress
            needDays = math.ceil(remainDays / speedsQueue[0])
            days += needDays

        count = 0
        while progressQueue and progressQueue[0] + speedsQueue[0] * days >= 100:
            progressQueue.popleft()
            speedsQueue.popleft()
            count += 1

        if count > 0:
            answer.append(count)

    return answer